File: app/src/main/java/nl/sjtek/client/android/cards/SonarrCard.java
package nl.sjtek.client.android.cards;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import java.util.List;
import java.util.Map;

import butterknife.BindView;
import butterknife.ButterKnife;
import nl.sjtek.client.android.R;
import nl.sjtek.control.data.responses.ResponseCollection;
import nl.sjtek.control.data.responses.SonarrResponse;

/**
 * Created by wouter on 15-11-16.
 */

public class SonarrCard extends BaseCard {

    private static final String EPISODE_NAME_TEMPLATE = "S%02dE%02d %s";
    private static final String DISK_TEMPLATE = "%02d%%";

    @BindView(R.id.textViewUpcoming)
    TextView textViewTitleUpcoming;
    @BindView(R.id.textViewShowTitle)
    TextView textViewShowTitle;
    @BindView(R.id.textViewEpisodeName)
    TextView textViewEpisodeName;
    @BindView(R.id.textViewDiskRootPercentage)
    TextView textViewDiskRoot;
    @BindView(R.id.textViewDiskDataPercentage)
    TextView textViewDiskData;

    public SonarrCard(Context context) {
        super(context);
    }

    public SonarrCard(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SonarrCard(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onShouldInflate(Context context) {
        inflate(context, R.layout.card_sonarr, this);
        ButterKnife.bind(this);
    }

    @SuppressLint("DefaultLocale")
    @Override
    protected void onUpdate(ResponseCollection update) {
        List<SonarrResponse.Episode> episodes = update.getSonarr().getUpcoming();
        Map<String, SonarrResponse.Disk> disks = update.getSonarr().getDiskUsage();

        boolean available = episodes.size() != 0;
        int visibility = available ? View.VISIBLE : View.GONE;
        textViewTitleUpcoming.setVisibility(visibility);
        textViewShowTitle.setVisibility(visibility);
        textViewEpisodeName.setVisibility(visibility);

        if (available) {
            SonarrResponse.Episode episode = episodes.get(0);
            textViewShowTitle.setText(episode.getSeriesTitle());
            textViewEpisodeName.setText(String.format(EPISODE_NAME_TEMPLATE, episode.getSeasonInt(), episode.getEpisodeInt(), episode.getEpisodeName()));
        }

        textViewDiskRoot.setText(getDiskPercentage(disks.get("/")));
        textViewDiskData.setText(getDiskPercentage(disks.get("/tv")));
    }

    @SuppressLint("DefaultLocale")
    private String getDiskPercentage(SonarrResponse.Disk disk) {
        return String.format(DISK_TEMPLATE, (int) ((disk.getTotal() - disk.getFree()) / disk.getTotal() * 100));
    }
}

File: app/src/main/java/nl/sjtek/client/android/cards/SonarrCard.kt
package nl.sjtek.client.android.cards

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import nl.sjtek.client.android.R
import nl.sjtek.control.data.parsers.ResponseHolder

/**
 * Card for displaying Sonarr information.
 */
class SonarrCard @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : BaseCard(context, attrs, defStyleAttr) {

    override fun onShouldInflate(context: Context) {
        inflate(context, R.layout.card_sonarr, this)
    }

    @SuppressLint("DefaultLocale")
    override fun onUpdate(update: ResponseHolder) {
//        if (update.sonarr == null || update.sonarr.diskUsage.isEmpty()) return
//        val episodes = update.sonarr.upcoming
//        val disks = update.sonarr.diskUsage
//
//        val available = episodes.isNotEmpty()
//        val visibility = if (available) View.VISIBLE else View.GONE
//        textViewUpcoming.visibility = visibility
//        textViewShowTitle.visibility = visibility
//        textViewEpisodeName.visibility = visibility
//
//        if (available) {
//            val episode = episodes[0]
//            textViewShowTitle.text = episode.seriesTitle
//            textViewEpisodeName.text = String.format(EPISODE_NAME_TEMPLATE, episode.seasonInt, episode.episodeInt, episode.episodeName)
//        }
//
//        textViewDiskRootPercentage.text = getDiskPercentage(disks["/"])
//        textViewDiskDataPercentage.text = getDiskPercentage(disks["/tv"])
    }

//    @SuppressLint("DefaultLocale")
//    private fun getDiskPercentage(disk: SonarrResponse.Disk?): String {
//        if (disk == null) return "n/a"
//        return "${((disk.total - disk.free) / disk.total * 100).toInt()}%"
//    }

    companion object {

        private val EPISODE_NAME_TEMPLATE = "S%02dE%02d %s"
    }
}
